src/BaseBlockTracker.ts

```diff
@@ -15,7 +15,7 @@ export interface Provider extends SafeEventEmitter {
 }
 
 interface BaseBlockTrackerArgs {
-  blockResetDuration?: number;
+  blockResetDuration?: number | undefined;
```

Comment:
This seems redundant. The `| undefined` is implied by the `?`

Reply:
Ah, this is a leftover from when I had temporarily upgraded to TS 4.4 as a part of some other changes I made that I ended up rolling back. But for now, you're right, I don't think this is necessary either.